optimum/habana/diffusers/pipelines/instantid/pipeline_stable_diffusion_xl_instantid.py

```diff
@@ -0,0 +1,1137 @@
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from math import ceil
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+import cv2
+import math
+
+import numpy as np
+import PIL.Image
+import torch
+
+from diffusers.image_processor import PipelineImageInput
+
+from diffusers.models import AutoencoderKL, ControlNetModel, UNet2DConditionModel
+
+from diffusers.utils import (
+    deprecate,
+)
+from diffusers.utils.torch_utils import is_compiled_module
+
+from transformers import (
+    CLIPImageProcessor,
+    CLIPTextModel,
+    CLIPTokenizer,
+    CLIPVisionModelWithProjection,
+    CLIPTextModelWithProjection
+)
+
+from diffusers import StableDiffusionXLControlNetPipeline
+from diffusers.pipelines.controlnet.multicontrolnet import MultiControlNetModel
+from diffusers.schedulers import KarrasDiffusionSchedulers
+
+from .ip_adapter.resampler import Resampler
+from .ip_adapter.utils import is_torch2_available
+
+if is_torch2_available():
+    from .ip_adapter.attention_processor import IPAttnProcessor2_0 as IPAttnProcessor, AttnProcessor2_0 as AttnProcessor
+else:
+    from .ip_adapter.attention_processor import IPAttnProcessor, AttnProcessor
+
+from optimum.utils import logging
+from ....transformers.gaudi_configuration import GaudiConfig
+from ....utils import HabanaProfile, speed_metrics
+from ..pipeline_utils import GaudiDiffusionPipeline
+from ..stable_diffusion_xl.pipeline_stable_diffusion_xl import GaudiStableDiffusionXLPipeline, GaudiStableDiffusionXLPipelineOutput
+from ..stable_diffusion.pipeline_stable_diffusion import retrieve_timesteps
+
+
+logger = logging.get_logger(__name__)
+
+
+class GaudiStableDiffusionXLInstantIDPipeline(GaudiDiffusionPipeline, StableDiffusionXLControlNetPipeline):
```

Comment:
Please extend `StableDiffusionXLInstantIDPipeline` and only override the methods that are hpu-specific

Reply:
Due to the impossibility of installing Instant ID for inheritance from [StableDiffusionXLInstantIDPipeline](https://github.com/InstantID/InstantID/blob/main/pipeline_stable_diffusion_xl_instantid.py#L136), it was decided to simply adapt the class to Gaudi by copying some of the methods and functions

Reply:
Sounds good.